src/payment/data/sagas.js

```diff
@@ -2,77 +2,99 @@ import { all, call, put, takeEvery } from 'redux-saga/effects';
 
 // Actions
 import {
-  FETCH_BASKET,
-  fetchBasketBegin,
-  fetchBasketSuccess,
-  fetchBasketFailure,
-  UPDATE_QUANTITY,
-  updateEnrollmentCodeQuantityBegin,
-  updateEnrollmentCodeQuantitySuccess,
-  updateEnrollmentCodeQuantityFailure,
+  basketDataReceived,
+  fetchBasket,
+  addCoupon,
+  removeCoupon,
+  updateQuantity,
 } from './actions';
 
 // Services
 import * as PaymentApiService from './service';
 
-import { saga as couponSaga, addCouponSuccess, addCouponBegin } from '../coupon';
 import { saga as cybersourceSaga } from '../cybersource';
 import { saga as payPalSaga } from '../paypal';
 import { handleErrors, handleMessages } from '../../feedback';
 
-export function* handleBasketResult(basketResult) {
-  yield put(fetchBasketSuccess(basketResult));
-  if (!basketResult.coupons || basketResult.coupons.length === 0) {
-    yield put(addCouponSuccess(null, null, null));
-  } else {
-    yield put(addCouponSuccess(
-      basketResult.coupons[0].id,
-      basketResult.coupons[0].code,
-      basketResult.coupons[0].benefitValue,
-    ));
+export function* handleFetchBasket() {
+  try {
+    yield put(fetchBasket.request());
+    const result = yield call(PaymentApiService.getBasket);
+    yield put(fetchBasket.success(result));
+    yield put(basketDataReceived(result));
+    yield call(handleMessages, result.messages, true);
+  } catch (error) {
+    yield put(fetchBasket.failure(error.message));
+    yield call(handleErrors, error, true);
+    if (error.basket) {
+      yield put(basketDataReceived(error.basket));
+    }
+  } finally {
+    yield put(fetchBasket.fulfill());
   }
 }
 
-export function* handleFetchBasket() {
-  yield put(fetchBasketBegin());
-  yield put(addCouponBegin());
+export function* handleAddCoupon({ payload }) {
   try {
-    const result = yield call(PaymentApiService.getBasket);
-    yield call(handleBasketResult, result);
+    yield put(addCoupon.request());
+    const result = yield call(PaymentApiService.postCoupon, payload.code);
+    yield put(addCoupon.success(result));
+    yield put(basketDataReceived(result));
+    yield call(handleMessages, result.messages, true);
+  } catch (error) {
+    yield put(addCoupon.failure(error.message));
+    yield call(handleErrors, error, true);
+    if (error.basket) {
+      yield put(basketDataReceived(error.basket));
+    }
+  } finally {
+    yield put(addCoupon.fulfill());
+  }
+}
+
+export function* handleRemoveCoupon({ payload }) {
+  try {
+    yield put(removeCoupon.request());
+    const result = yield call(PaymentApiService.deleteCoupon, payload.code);
```

Comment:
Nit: rename `deleteCoupon` to `removeCoupon` to keep consistent naming.

Reply:
We have been using a pattern of naming where UI focused work is named after what's happening and functions that make api calls are named with HTTP verbs.